packages/components/src/navigation/index.js

```diff
@@ -50,9 +55,15 @@ export default function Navigation( {
 		}
 	}, [] );
 
+	// Used to prevent excessive useEffect fires when navigation is being controlled by parent component
+	const controlledMenuUpdate = useRef( { setActiveMenu, menu } );
+	useLayoutEffect( () => {
```

Comment:
Just to [make sure it's always up to date.](https://kentcdodds.com/blog/useeffect-vs-uselayouteffect#one-special-case). If I'd used a `useEffect` here, and then some other future change tried to use this hook from a `useLayoutEffect`, that would be a stale reference because `layoutEffects` fire first.

Updating the ref with `useLayoutEffect` here should ensure it's always up to date, regardless of what kind of effect tries to call it later.